src/webgpu/api/validation/buffer/destroy.spec.ts

```diff
@@ -56,13 +56,14 @@ g.test('twice')
 
 g.test('while_mapped')
   .desc(
-    `Test destroying buffers while mapped.
+    `Test destroying buffers while mapped or after being unmapped.
       - Tests {mappable, unmappable mapAtCreation, mappable mapAtCreation}
-      - Tests while {mapped, mapped at creation}`
+      - Tests while {mapped, mapped at creation, unmapped}`
   )
   .paramsSubcasesOnly(u =>
     u //
       .combine('mappedAtCreation', [false, true])
+      .combine('unmapBeforeDestoy', [false, true])
       .combineWithParams([
         { usage: GPUConst.BufferUsage.COPY_SRC },
```

Comment:
Cases where it's mappable without actually being mapped (in case the presence of a usage flag changes the implementation of mappedAtCreation)
```suggestion
        { usage: GPUConst.BufferUsage.COPY_SRC },
        { usage: GPUConst.BufferUsage.MAP_WRITE | GPUConst.BufferUsage.COPY_SRC },
        { usage: GPUConst.BufferUsage.COPY_DST | GPUConst.BufferUsage.MAP_READ },
```